i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -229,4 +425,573 @@ mod tests {
+                 # Foo\n\n\
+                 - [The Foo Chapter](foo.md)\n\
+                 \t- [The Bar Section](foo/bar.md)\n\
+                 \t\t- [The Baz Subsection](foo/bar/baz.md)\n\
+                 - [Foo Exercises](exercises/foo.md)",
+            ),
+            (
+                "src/index.md",
+                "# Intro to X\n\
+                 \n\
+                 In this course, you will learn X.\n",
+            ),
+            (
+                "src/foo.md",
+                "# How to Foo\n\
+                 \n\
+                 The first paragraph about Foo.\n",
+            ),
+            (
+                "src/foo/bar.md",
+                "# How to Bar\n\
+                 \n\
+                 The first paragraph about Bar.\n",
+            ),
+            (
+                "src/foo/bar/baz.md",
+                "# How to Baz\n\
+                 \n\
+                 The first paragraph about Baz.\n",
+            ),
+            (
+                "src/exercises/foo.md",
+                "# Exercises on Foo\n\
+                 \n\
+                 Try out this Foo example.\n",
+            ),
+        ])?;
+
+        let catalogs = create_catalogs(&ctx, 3)?;
+
+        let message_tuples = catalogs
+            .into_iter()
+            .map(|(file_rel_path, catalog)| (file_rel_path, msgids_from_catalog(catalog)))
+            .collect::<Vec<(PathBuf, Vec<String>)>>();
+
+        let expected_message_tuples = HashMap::from([
+            (
+                PathBuf::from("summary/summary"),
+                vec![
+                    "Intro",
+                    "Foo",
+                    "The Foo Chapter",
+                    "The Bar Section",
+                    "The Baz Subsection",
+                    "Foo Exercises",
+                ],
+            ),
+            (
+                PathBuf::from("summary/intro"),
+                vec!["Intro to X", "In this course, you will learn X."],
+            ),
+            (
+                PathBuf::from("foo/the-foo-chapter"),
+                vec!["How to Foo", "The first paragraph about Foo."],
+            ),
+            (
+                PathBuf::from("foo/the-foo-chapter/the-bar-section"),
+                vec![
+                    "How to Bar",
+                    "The first paragraph about Bar.",
+                    "How to Baz",
+                    "The first paragraph about Baz.",
+                ],
+            ),
+            (
+                PathBuf::from("foo/foo-exercises"),
+                vec!["Exercises on Foo", "Try out this Foo example."],
+            ),
+        ]);
+
+        // The message ids in a catalog do not have a constant order,
+        // so check for the presence of each one in the vector.
+        assert_eq!(expected_message_tuples.keys().len(), message_tuples.len());
+        for (file_path, msgids) in message_tuples {
+            let expected_msgids = expected_message_tuples.get(&file_path).unwrap();
```

Comment:
Doing `map.get("key").unwrap()` is the same as `map["key"]` so I would prefer the shorter version.

Reply:
Done